### ISSUE TITLE
Failed to compile for `wasm32-unknown-emscripten` because of `zune_jpeg`

### ISSUE THREAD

**Issue by pixelpicosean** [linebender/resvg#851]
It seems like `zune_jpeg` causes WASM target failed to compile, anyone has same problem?

It keep saying `emcc` not found, then I activated my local emsdk the following message is printed:
```sh
wasm-ld: error: emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libstandalonewasm.a(__main_void.o): undefined symbol: main
```

**Comment by DJMcNab** [comment]
I don't expect us to support emscripten - are you able to use `wasm32-unknown-unknown` instead? You might need to use wasm-bindgen to make that work.

**Comment by waywardmonkeys** [comment]
Are you building a library or an application?

I think you need to build the application and then it'll have a `main` and perhaps work.

I'm not sure why it is trying to link a standalone wasm file for `zune_jpeg`

If you build without default features, then it does at least build:

```
cargo build --target wasm32-unknown-emscripten --no-default-features -p resvg
```

So it does seem like perhaps something perhaps specific to zune-jpeg.

**Comment by pixelpicosean** [comment]
I build resvg as a static lib, and been using it for a few months. Targeting both native and wasm without any problems until `zune_jpeg` appears.

Thanks anyway, I've manually changed it back to `jpeg_encoder` and it works again.

I will close the issue if no one else cares about wasm targeting, let me know if you have ideas.

**Comment by waywardmonkeys** [comment]
I'd try seeing if this happens with just `zune_jpeg` by itself and if so, reporting it there as well.

**Comment by kleisauke** [comment]
This issue is likely related to https://github.com/etemesi254/zune-image/pull/242. You can address it using the following workaround:
https://github.com/kleisauke/wasm-vips/blob/3f143eca218043d53dcd17b765419731030c5075/build.sh#L450-L457

**Comment by waywardmonkeys** [comment]
Ah, yes, that would cause that problem!

Thank you, @kleisauke.

I'll close this since there's no issue within `resvg` here and hopefully, `zune_jpeg` will get a new release soon.

**Comment by kleisauke** [comment]
For reference, this issue was resolved in zune-jpeg 0.4.14. Running `cargo update` will incorporate this.
https://github.com/kleisauke/wasm-vips/blob/f3255fb1dfef37ec3d665b37794dbb8dc4a6f8d5/build.sh#L457-L459
